tests/test_load_data.py

```diff
@@ -57,6 +58,16 @@ def test_roundtrip_other_dir(test_data_dir, tmp_path):
     assert dsnpy.nfields == 1
 
 
+def test_roundtrip_current_dir(test_data_dir):
+    os.chdir(test_data_dir / "idefix_spherical_planet3d")
+    gf = GasDataSet(500)["RHO"].azimuthal_average()
+    gf.save()
+    dsnpy = GasDataSet.from_npy(500, operation="azimuthal_average")
+    assert dsnpy.nfields == 1
+    shutil.rmtree("rho")
+    shutil.rmtree("header")
```

Comment:
this will pollute the repo, making the test's results potentially dependent of external state. Instead, I suggest moving to a temporary dir, make sure that there's a `idefi.ini` file in there. Also, let's keep the test focused and not do a full round trip.

```suggestion
def test_save_with_existing_data(test_data_dir, tmp_path):
    directory = test_data_dir / "idefix_spherical_planet3d"
    gf = GasDataSet(500, directory=directory)["RHO"].azimuthal_average()
	shutil.copy(directory / "idefix.ini", tmp_path / "idefix.ini")
	gf.save()
```

Reply:
I agree it's polluting the repo! But I see 2 problems in your suggested change. First, you do not use the `tmp_path` directory when saving the density field. Second, if you write `gf.save(tmp_path)`, you do not capture the problem as I think it appears when the saving directory is the same as the data directory.

Reply:
> Second, if you write gf.save(tmp_path), you do not capture the problem as I think it appears when the saving directory is the same as the data directory.

That's indeed what I meant to type, and I still think it would show the problem ?

Reply:
> I agree it's polluting the repo, that is why I tried to delete the created files with rmtree. 

As you wrote it, this only works if the test completes. You still get pollution if it fails mid-air. What's worse, in a very broken state, the test could actually remove data that was produced in a different run.